src/main/java/com/deavensoft/paketomat/city/CityController.java

```diff
@@ -0,0 +1,126 @@
+        this.mapper = mapper;
+    }
+
+    @GetMapping
+    @Operation(summary = "Get cities", description = "Get all cities")
+    @ApiResponse(responseCode = "200", description = "All cities are returned")
+    public List<City> getAllCities() {
+        return cityServiceImpl.getAllCities();
+    }
+
+    @PostMapping
+    @Operation(summary = "Add new city", description = "Add new city to the database")
+    @ApiResponse(responseCode = "200", description = "New city added")
+    public void save(@RequestBody City c) {
+        cityServiceImpl.save(c);
+    }
+
+    @GetMapping(path = "/{id}")
+    @Operation(summary = "Get city", description = "Get city with specified id")
+    @ApiResponse(responseCode = "200", description = "City with specified id returned")
+    public Optional<City> findCityById(@PathVariable(name = "id") long id) {
+        return cityServiceImpl.findCityById(id);
+    }
+
+    @DeleteMapping(path = "/{id}")
+    @Operation(summary = "Delete city", description = "Delete city with specified id")
+    @ApiResponse(responseCode = "200", description = "City with specified id deleted")
+    public int deleteCity(@PathVariable(name = "id") Long id) {
+        Optional<City> c = findCityById(id);
+        if (c.isEmpty())
+           log.info("there is no city with id:" + id);
+        if(!c.isEmpty()) {
+            cityServiceImpl.deleteCity(c.get());
+            log.info("City deleted");
+        }
+        return 1;
+    }
+
+    @GetMapping(path = "/check")
+    public String checkCities() throws IOException {
+        OkHttpClient client = new OkHttpClient();
+
+        Request request = new Request.Builder()
+                .url("https://countries-cities.p.rapidapi.com/location/country/RS/city/list?page=3&per_page=100&population=1501")
+                .get()
+                .addHeader("X-RapidAPI-Key", "c427cbdd50msh072fe84a375af9cp1f2596jsn0f5919dacf12")
+                .addHeader("X-RapidAPI-Host", "countries-cities.p.rapidapi.com")
+                .build();
+
+        Response response = client.newCall(request).execute();
+
+
+        CitiesDto citiesDto = new ObjectMapper().readValue(response.body().string(), CitiesDto.class);
+        List<CityDto> cities = citiesDto.getCities();
+        Integer totalNum = citiesDto.getTotalPages();
+        Integer numCities = citiesDto.getTotalCities();
+        Integer numFromTable = getAllCities().size();
+        client.cancel(response);
+        if (numFromTable < numCities) {
+
+            for (CityDto city : cities) {
+
+                save(mapper.cityDtoToCity(city));
+            }
+
+            for (int i = 2; i <= totalNum; i++) {
+                String ii = Integer.toString(i);
+                String url = "https://countries-cities.p.rapidapi.com/location/country/RS/city/list?page={1}&per_page=100&population=1501";
+                String newUri = url.replace("{1}", ii);
+
+                Request requestPaginate = new Request.Builder()
```

Comment:
Please find and replace code duplicates (74-81 and 101-108).

Reply:
Done.